specs/core/0_beacon-chain.md

```diff
@@ -749,23 +761,32 @@ def add_validator(state: State,
                      sig=proof_of_possession,
                      domain=get_domain(state, current_slot, DOMAIN_DEPOSIT))
     # Pubkey uniqueness
-    assert pubkey not in [v.pubkey for v in state.validators]
-    rec = ValidatorRecord(
-        pubkey=pubkey,
-        withdrawal_credentials=withdrawal_credentials,
-        randao_commitment=randao_commitment,
-        randao_skips=0,
-        balance=DEPOSIT_SIZE * GWEI_PER_ETH,
-        status=status,
-        last_status_change_slot=current_slot,
-        exit_seq=0
-    )
-    index = min_empty_validator(state.validators)
-    if index is None:
-        state.validators.append(rec)
-        return len(state.validators) - 1
+    validator_pubkeys = [v.pubkey for v in state.validators]
+    if pubkey not in validator_pubkeys:
+        assert deposit_size == DEPOSIT_SIZE
+        rec = ValidatorRecord(
+            pubkey=pubkey,
+            withdrawal_credentials=withdrawal_credentials,
+            randao_commitment=randao_commitment,
+            randao_skips=0,
+            balance=DEPOSIT_SIZE * GWEI_PER_ETH,
+            status=status,
+            last_status_change_slot=current_slot,
+            exit_seq=0
+        )
+        index = min_empty_validator(state.validators)
+        if index is None:
+            state.validators.append(rec)
+            return len(state.validators) - 1
+        else:
+            state.validators[index] = rec
+            return index
     else:
```

Comment:
Should probably make sure that the validator represented by the pub_key is not `WITHDRAWN`

Reply:
Just fixed with https://github.com/ethereum/eth2.0-specs/pull/162/commits/9a20ecb9e46817bea17637193ca866d976ff952b

Reply:
This now bypasses the deletion period.

The `deposit_data.msg_value == DEPOSIT_SIZE` conditional in `DEPOSIT_PROOF` validity also needs to be reworked

Reply:
> This now bypasses the deletion period.

Aah, actually you're right, that last commit would have made it possible to use the same receipt to deposit twice. Just edited to fix.

There is no longer a `deposit_data.msg_value == DEPOSIT_SIZE` conditional in `DEPOSIT_PROOF`.